File: Clase 7/PYTHON/ej3/main.py


def imprimir_matriz(m):
    for i in range(3):
        for j in range(3):
            print(m[i][j], end=" ")
        print(" ")

def ingreso_matriz(m):
    for i in range(3):
        for j in range(3):
            m[i][j] = int(input(f"Ingrese el valor de [{i},{j}]"))

def verificar_cubo_magico(m):
    suma_f0 = 0
    suma_f1 = 0
    suma_f2 = 0
    suma_c0 = 0
    suma_c1 = 0
    suma_c2 = 0
    suma_d1 = 0
    suma_d2 = 0

    for i in range(3):
        for j in range(3):
            #filas
            if i == 0: suma_f0 += m[i][j]
            if i == 1: suma_f1 += m[i][j]
            if i == 2: suma_f2 += m[i][j]

            #columnas
            if j == 0: suma_c0 += m[i][j]
            if j == 1: suma_c1 += m[i][j]
            if j == 2: suma_c2 += m[i][j]

            #diagonales
            if i == j: suma_d1 += m[i][j]
            if i + j == 2: suma_d2 += m[i][j]
    
    if suma_f0 == suma_f1 and suma_f1 == suma_f2 and suma_f2 == suma_c0 and suma_c0 == suma_c1 and suma_c1 == suma_c2 and suma_c2 == suma_d1 and suma_d1 == suma_d2:
        print("Si, el cubo es magico!!!")
    else:
        print("No, no es magico")


if __name__ == "__main__":
    # declaracion explicita
    # cubo = [[1,2,3],[4,5,6],[7,8,9]]
    #cubo = [[,,],[,,],[,,]]

    #Declaracion matriz vacia
    #cubo = []
    #for i in range(3):
    #    cubo.append([0]*3)
    
    # Declaracion simple
    columnas = 3
    filas = 3
    cubo = [[0] * columnas for i in range(filas)]

    ingreso_matriz(cubo)
    imprimir_matriz(cubo)
    verificar_cubo_magico(cubo)

File: Clase 7/JAVA/ej3/Main.java
import java.util.Scanner;

public class Main{

    public static void ingresarMatriz(int[][] m){
        Scanner lector = new Scanner(System.in);
        for(int i = 0; i < 3; i++){//para filas
            for(int j = 0; j < 3; j++){//para columnas
                System.out.print("Ingrese el valor de la celda ["+i+"]["+j+"]");
                m[i][j] = lector.nextInt();
            }
        }
    }

    public static void imprimirMatriz(int[][] m){
        for(int i = 0; i < 3; i++){//para filas
            for(int j = 0; j < 3; j++){//para columnas
                System.out.print(m[i][j] + " ");
            }
            System.out.println(" ");
        }
    }

    public static void verificarCuboMagico(int[][] m){
        int sumaGlobal = 0;
        int sumaFila = 0;
        int sumaColumna = 0;
        int sumaDiagonal = 0;

        //Para Filas
        for(int i = 0; i < 3; i++){//para filas
            for(int j = 0; j < 3; j++){//para columnas
                sumaFila += m[i][j];
            }
            if(sumaGlobal == 0)
                sumaGlobal = sumaFila;
            if(sumaFila != sumaGlobal){
                System.out.println("El cubo no es magico");
                return;
            }
            sumaFila = 0;
        }
        //Para Columnas
        for(int j = 0; j < 3; j++){//para filas
            for(int i = 0; i < 3; i++){//para columnas
                sumaColumna += m[i][j];
            }
            if(sumaGlobal == 0)
                sumaGlobal = sumaColumna;
            if(sumaColumna != sumaGlobal){
                System.out.println("El cubo no es magico");
                return;
            }
            sumaColumna = 0;
        }
        //Para diagonales
        for(int i = 0; i < 3; i++){//para filas
            for(int j = 0; j < 3; j++){//para columnas
                if(i == j)
                    sumaDiagonal += m[i][j];
            }
        }
        if(sumaGlobal == 0)
            sumaGlobal = sumaDiagonal;
        if(sumaDiagonal != sumaGlobal){
            System.out.println("El cubo no es magico");
            return;
        }
        sumaDiagonal = 0;

        //Para diagonales
        for(int i = 0; i < 3; i++){//para filas
            for(int j = 0; j < 3; j++){//para columnas
                if(i + j == 2)
                    sumaDiagonal += m[i][j];
            }
        }
        if(sumaGlobal == 0)
            sumaGlobal = sumaDiagonal;
        if(sumaDiagonal != sumaGlobal){
            System.out.println("El cubo no es magico");
            return;
        }
        sumaDiagonal = 0;
        //Conclusion de la Funcion
        System.out.println("El Cubo es Magico!!!");
    }

    public static void verificarCuboMagico2(int[][] m){
        int sumaFila0 = 0;
        int sumaFila1 = 0;
        int sumaFila2 = 0;
        int sumaColumna0 = 0;
        int sumaColumna1 = 0;
        int sumaColumna2 = 0;
        int sumaDiagonal1 = 0;
        int sumaDiagonal2 = 0;

        //Para Filas
        for(int i = 0; i < 3; i++){//para filas
            for(int j = 0; j < 3; j++){//para columnas
                //filas
                if(i == 0) sumaFila0 += m[i][j];
                if(i == 1) sumaFila1 += m[i][j];
                if(i == 2) sumaFila2 += m[i][j];

                //columnas
                if(j == 0) sumaColumna0 += m[i][j];
                if(j == 1) sumaColumna1 += m[i][j];
                if(j == 2) sumaColumna2 += m[i][j];

                //diagonales
                if(i == j) sumaDiagonal1 += m[i][j];
                if(i + j == 2) sumaDiagonal2 += m[i][j];
            }
        }
        if(sumaFila0 != sumaFila1 || sumaFila0 != sumaFila2 || sumaFila1 != sumaFila2 || sumaFila0 != sumaColumna0 || sumaColumna0 != sumaColumna1 || sumaColumna0 != sumaColumna2 || sumaColumna1 != sumaColumna2 || sumaColumna0 != sumaDiagonal1 || sumaDiagonal1 != sumaDiagonal2){
            System.out.println("El cubo no es magico");
            return;
        }
        //Conclusion de la Funcion
        System.out.println("El Cubo es Magico!!!");
    }


    public static void main(String[] args){

        // explicita
        //int[][] cubo = {{1,2,3},{4,5,6},{7,8,9}};
        // declaracion  de matriz en java con valor de dimension
        int[][] cubo = new int[3][3];
        ingresarMatriz(cubo);
        imprimirMatriz(cubo);
        verificarCuboMagico2(cubo);
    }
}